src/web/mod.rs

```diff
@@ -50,8 +54,12 @@ use url::form_urlencoded;
 const FRAGMENT: &AsciiSet = &CONTROLS.add(b' ').add(b'"').add(b'<').add(b'>').add(b'`');
 const PATH: &AsciiSet = &FRAGMENT.add(b'#').add(b'?').add(b'{').add(b'}');
 
-pub(crate) fn encode_url_path(path: &str) -> String {
-    utf8_percent_encode(path, PATH).to_string()
+/// encode a path & parse it for usage in an Uri or CanonicalUrl
+pub(crate) fn encode_path_for_uri<P: AsRef<str>>(path: P) -> Result<uri::PathAndQuery, Error> {
+    utf8_percent_encode(path.as_ref(), PATH)
+        .to_string()
+        .try_into()
+        .context("error parsing URI PathAndQuery from path")
```

Comment:
I don't understand what changed here; how is `utf8_percent_encode` returning a Result when it returned `String` before? I don't see changes to it in this PR.

Reply:
`.try_into::<uri::PathAndQuery>()` is where the result is coming from, since the new encode method is not returning a `String` any more